### PR TITLE
Changes nf-core cache dir to user's home

### LINKED ISSUE #379: problem with nfcore_cache.sqlite within /tmp
Hi all,
I think will be a nice idea to have the nfcore_cache.sqlite within a subfolder in tmp because if two users use the program at the same time the privileges will prevent to use the tool.

For example I cannot even use nf-core --help 

Luca

### PR DESCRIPTION
**Pull request by sven1103**
Fixes #379 .

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/tools/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by apeltzer** [comment]
The `import temfile` can be dropped according to: 

https://lgtm.com/projects/g/nf-core/tools/rev/pr-78a866159bbc8065b95d1ea9b362de23429a32d8

File: nf_core/utils.py [hunk 1]
```diff
@@ -75,15 +75,13 @@ def setup_requests_cachedir():
     """
     # Only import it if we need it
     import requests_cache
+    
 
-    cachedir = os.path.join(tempfile.gettempdir(), 'nfcore_cache')
+    cachedir = os.path.join(os.getenv("HOME"), '.nfcore_cache')
```
Comment (ewels) [inline nf_core/utils.py]: What do you think about nesting this inside the `~/.nextflow` home directory? One fewer home directory.. (even if it is hidden)

File: nf_core/utils.py [hunk 2]
```diff
@@ -75,15 +75,13 @@ def setup_requests_cachedir():
     """
```
Comment (ewels) [inline nf_core/utils.py]: Function description needs updating as well

**Comment by ewels** [comment]
In fact, instead of using the user's home directory, can we not just add in a user-specific subdir on the temp path? Needs to be persistent between runs though. Maybe [`os.getlogin()`](https://docs.python.org/3/library/os.html#os.getlogin)?

```python
os.path.join(tempfile.gettempdir(), os.getlogin(), 'nfcore_cache')
```

File: nf_core/utils.py (hunk 1 above)
Comment (sven1103) [reply nf_core/utils.py]: I would stick to the common application centric approach: hidden cache dirs and config dirs for each application. Nf-core is a different application then Nextflow.

**Comment by sven1103** [comment]
> In fact, instead of using the user's home directory, can we not just add in a user-specific subdir on the temp path? Needs to be persistent between runs though. Maybe [`os.getlogin()`](https://docs.python.org/3/library/os.html#os.getlogin)?
> 
> ```python
> os.path.join(tempfile.gettempdir(), os.getlogin(), 'nfcore_cache')
> ```

I think the home dir is a better solution, because the tmp folder might not be persistent. Depending on the system setup, it might get deleted regularly. And we do not want that for the request cache imo :)

File: nf_core/utils.py (hunk 1 above)
Comment (ewels) [reply nf_core/utils.py]: How about `.nfcore/cache/` instead then? Future-proofs us for adding more stuff later..?

Comment (apeltzer) [reply nf_core/utils.py]: Would be a good idea I think 👍

**Comment by sven1103** [comment]
@ewels better?

**Comment by apeltzer** [comment]
we're reviewing
@sven1103  you broke the tests!

**Pull request merged**